### PR TITLE
add Makefile and convert many of the manual tasks

### LINKED ISSUE #7: Add Makefiles for recurring tasks
So that people can test better remotely.

### PR DESCRIPTION
**Pull request by imrehg**
Currently handling:

* Plotly export generation
* Clean notebooks generation

Closes #7

### CODE REVIEW

File: .github/workflows/site-deploy.yml [hunk 1]
```diff
@@ -24,12 +24,9 @@ jobs:
...
+        working-directory: src/
         run: |
-          jupyter nbconvert --to python adult.ipynb
-          python3 adult.py
-          mkdir -p ../site/src/figures/
-          mv *.json ../site/src/figures
+          make figures
 
       - name: Build adversarial plots (temporary)
         working-directory: src/notebooks/finance
```
Comment (tcbegley) [inline .github/workflows/site-deploy.yml]: I think maybe this can be removed now?

File: .github/workflows/site-preview.yml [hunk 2]
```diff
@@ -36,12 +38,9 @@ jobs:
...
-        working-directory: src/notebooks
+        working-directory: src/
         run: |
-          jupyter nbconvert --to python adult.ipynb
-          python3 adult.py
-          mkdir -p ../site/src/figures/
-          mv *.json ../site/src/figures
+          make figures
 
       - name: Build adversarial plots (temporary)
```
Comment (tcbegley) [inline .github/workflows/site-preview.yml]: ditto

**Review by tcbegley — CHANGES_REQUESTED** [review]
I think some of the temporary changes I made to the workflows can be stripped out now (sorry)

File: .github/workflows/site-preview.yml (hunk 2 above)
Comment (imrehg) [reply .github/workflows/site-preview.yml]: Checking, sorry, slipped through.

File: .github/workflows/site-deploy.yml (hunk 1 above)
Comment (imrehg) [reply .github/workflows/site-deploy.yml]: Checking, sorry, slipped through.

**Comment by tcbegley** [comment]
Made a couple of changes, first is ignoring notebook checkpoints in figure creation which was breaking local builds (due to breaking relative paths). Second is updating the paths to the generated JSON files in the Mdx content.

~Running locally I was having some issues with `make figures`, specifically the call to `gcp` was failing telling me that `src/site/src/figures` didn't exist (it does). Is it a macOS thing? Seemed that that step was ok before. Let's see what happens to the build.~ nvm, saw your comment in the code, resolved with `brew install coreutils`

**Review by tcbegley — APPROVED** [review]
lgtm

**Pull request merged**